builtin/mainmenu/textures.lua

```diff
@@ -145,6 +145,28 @@ function mm_texture.set_game(identifier,gamedetails)
 	
 	local path = gamedetails.path .. DIR_DELIM .."menu" ..
 									 DIR_DELIM .. identifier .. ".png"
+
+	local count = 1
+	for i = 9, 1, -1 do
+		local check_path = gamedetails.path .. DIR_DELIM .."menu" ..
+			DIR_DELIM .. identifier .. "." .. i .. ".png"
+		local file = io.open(check_path, "rb")
+		if file ~= nil then
+			count = i
+			io.close(file)
+			break
+		end
+	end
+	local new_path = gamedetails.path .. DIR_DELIM .."menu" .. DIR_DELIM ..
+		identifier .. "." .. math.random(1, count) .. ".png"
+	local exists = io.open(new_path, "rb")
+	if exists ~= nil and math.random(1, (count+1)) ~= 1 then
```

Comment:
Your code correctly assigns 25% prob. to every image (if you have 4 total) but is very hard to understand IMO
what you should do is generate a numberbetween `0` and `count` and use background.png if the number is `0`